docs/src/explanations/printing.md

```diff
@@ -103,6 +103,8 @@ Message:
 
 Notice the use of `+` between `cf` interpolated "strings". The results of `cf` interpolation can be concatenated by using the `+` operator.
 
+> :warning: The Scala s-interpolator is deprecated since 3.5 and will output the warning: "s-interpolators for Chisel assert, assume and printf statements are deprecated (since 3.5); use p or cf interpolators instead"
```

Comment:
Can you do an actual execution driven example rather than just a text warning?

Since it fails in this PR, you can do the following:
````
```scala mdoc:fail
class MyModule extends Module {
  val in = IO(Input(UInt(8.W)))
  printf(s"in = $in\n")
}
```
````
This should fail to compile and render with the exact error when you run `sbt docs/mdoc`